### PR TITLE
fix(curriculum): improve seo with better-used terms

### PR DESCRIPTION
**Pull request by RawandDev**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

**Note**: In most of the content, it says **programming**, but in some cases that I edited was written as **programing**, so I 
changed them to **programming**. I know that both of them are correct, but it is better to modify them to only one word.

### CODE REVIEW

File: client/i18n/locales/english/intro.json [hunk 1]
```diff
@@ -82,10 +82,10 @@
     "title": "JavaScript Algorithms and Data Structures",
     "intro": [
       "While HTML and CSS control the content and styling of a page, JavaScript is used to make it interactive. In the JavaScript Algorithm and Data Structures Certification, you'll learn the fundamentals of JavaScript including variables, arrays, objects, loops, and functions.",
-      "Once you have the fundamentals down, you'll apply that knowledge by creating algorithms to manipulate strings, factorialize numbers, and even calculate the orbit of the International Space Station.",
-      "Along the way, you'll also learn two important programing styles or paradigms: Object Oriented Programing (OOP), and Functional Programing (FP)."
+      "Once you have the fundamentals down, you'll apply that knowledge by creating algorithms to manipulate strings, factorize numbers, and even calculate the orbit of the International Space Station.",
+      "Along the way, you'll also learn two important programming styles or paradigms: Object-Oriented Programming (OOP) and Functional Programming (FP)."
```
Comment (ShaunSHamilton) [inline client/i18n/locales/english/intro.json:86]:
```suggestion
      "Once you have the fundamentals down, you'll apply that knowledge by creating algorithms to manipulate strings, factorialize numbers, and even calculate the orbit of the International Space Station.",
      "Along the way, you'll also learn two important programming styles or paradigms: Object Oriented Programming (OOP) and Functional Programming (FP)."
```

File: client/i18n/locales/english/intro.json [hunk 2]
```diff
@@ -128,15 +128,15 @@
...
+          "To write an effective algorithm, it helps to break a problem down into smaller parts and think carefully about how to solve each part with code.",
           "In this course, you'll learn the fundamentals of algorithmic thinking by writing algorithms that do everything from converting temperatures to handling complex 2D arrays."
         ]
       },
       "object-oriented-programming": {
         "title": "Object Oriented Programming",
         "intro": [
-          "OOP, or Object Oriented Programming, is one of the major approaches to the software development process. In OOP, objects and classes are used to organize code to describe things and what they can do.",
-          "In this course, you'll learn the basic principles of OOP in JavaScript including the <code>this</code> keyword, prototype chains, constructors, and inheritance."
+          "OOP, or Object-Oriented Programming, is one of the major approaches to the software development process. In OOP, objects and classes organize code to describe things and what they can do.",
```
Comment (ShaunSHamilton) [inline client/i18n/locales/english/intro.json:138]:
```suggestion
          "OOP, or Object Oriented Programming, is one of the major approaches to the software development process. In OOP, objects and classes organize code to describe things and what they can do.",
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Thank you, for taking the time to contribute. Just some recommended changes:
- `factorialize` is not a typo - just the verb form of factorial
- Object Oriented is more prominent than Object-Oriented

Whilst we do care about grammar, and, personally, I do not like perpetuating misnomers, if no one can find the content, because they are using a different form/term, then the grammar argument comes second.

**Comment by RawandDev** [comment]
@ShaunSHamilton Thanks for the clarification. Is it better now?

**Review by ShaunSHamilton — APPROVED** [review]
Thanks, for making those changes. This LGTM (looks good to me) 👍 

---

For future PRs, please take a loot at the recommended best-practice for commit and PR naming: https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title

**Comment by RawandDev** [comment]
Thank you for providing more information about PRs. This is my first PR, and I will try to contribute more. I'm sorry if something didn't go well.

**Review by Sembauke — APPROVED** [review]

**Pull request merged**